### ISSUE TITLE
[no-standalone-expect]: false positive for `expect.any` in `__mocks__` file

### ISSUE THREAD

**Issue by garyking** [jest-community/eslint-plugin-jest#1189]
I have a file in `__mocks__` called `twitch-js-api.js` and I'm using the following:

```js
const formatStreams = jest.fn().mockReturnValue([
  {
    channel: {
      logo: expect.any(String)
    }
  }
])
```

Yet this rule now warns on `expect.any(String)`? Isn't `expect` allowed in mock files? This code has worked for years.

**Comment by G-Rath** [comment]
Technically the rule is correct, and the fact that it wasn't erroring before was arguably a bug but I'm thinking maybe we should actually ignore `except` member calls except for `hasAssertions` and `assertions` as everything is valid to be used outside of test blocks 🤔

**Comment by G-Rath** [comment]
I'm going to close this as a duplicate of #1186
